### ISSUE TITLE
Metadata file System.Runtime.dll could not be found

### ISSUE THREAD

**Issue by MelbourneDeveloper** [aspnet/RoslynCodeDomProvider#65]
@Jinhuafei 

We use the VB and C# Code Dom Providers in our ASP.Net Core app (.NET Framework). The issue is that when we attempt to compile using the provider, it looks for System.Runtime.dll in an almost random location. Every time we go to compile, the location shifts to a new location. So, we have to continuously create folders and move System.Runtime.dll in there. It doesn't matter if we include a reference to it or not. It's obvious that what is happening is that the compiler thinks it has made a temp copy of System.Runtime.dll in a given system folder, and then tries to reference it, but in fact, it hasn't made the copy so the compilation just fails. However, if we copy the DLL in to the folder that it thinks exists, the compilation works fine.

The error is this:

> Compiler Error Number: CS0006
> Error Text: Metadata file 'C:\Windows\system32\config\systemprofile\AppData\Local\assembly\dl3\8GTGDQKQ.4QR\V42JN801.7NM\36a4366e\f1f27652_1972d301\System.Runtime.dll' could not be found

The folder C:\Windows\system32\config\systemprofile\AppData\Local\assembly\dl3\8GTGDQKQ.4QR\V42JN801.7NM\36a4366e is piling up with folders we've had to move copies of the DLL in to.

I believe that the version is Microsoft.CodeDom.Providers.DotNetCompilerPlatform : 1.0.8

This is a continuation of the bug logged here:
https://github.com/dotnet/roslyn/issues/24630

You mentioned we should use Fuslogvw.exe, but this tool is just a log viewer. How can we turn logging on? And where will the log be located so that we can get you the information you need to fix the bug?

@acecoderlaura

**Comment by Jinhuafei** [comment]
@MelbourneDeveloper you are using DotNetCompilerPlatform nupkg in your asp.net core app to dynamically generate/compile your code? If so, the problem may be caused by [this line of code](https://github.com/aspnet/RoslynCodeDomProvider/blob/12975a5a1616dbefac02ba8113204b59cba7eeec/src/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/CSharpCompiler.cs#L114). You can comment it out and give it try to see if it's the root cause.

**Comment by MelbourneDeveloper** [comment]
@Jinhuafei yes. That certainly looks like the bug. I will try it out.

**Comment by AceCoderLaura** [comment]
When is this bug going to get fixed? How do I track the progress of bug 913691? This bug is serious and I can't even downgrade System.Runtime to 4.0.0.0 in my project to pander to this hack! I downgraded the reference in my solution but version 4.1.2.0 is outputted to the bin folder. I've tried to fix my build's reference resolution with a binding redirect but it doesn't work! What can I do without forking and fixing this issue myself? Can you see what's wrong with my binding redirect?

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>

  <system.web>
    <!-- Session Timeout in Minutes - IIS Only -->
    <sessionState timeout="1440"/>
    <httpRuntime maxRequestLength="2147483647" />
  </system.web>

  <system.webServer>

    <handlers>
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>

    <aspNetCore processPath=".\Adapt.Services.exe" arguments="" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="false" />

  </system.webServer>

  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <assemblyIdentity name="System.Runtime" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
      <bindingRedirect oldVersion="0.0.0.0-4.1.2.0" newVersion="4.0.0.0" />
    </assemblyBinding>
  </runtime>

</configuration>
```

**Comment by AceCoderLaura** [comment]
TargetFramework is net461.
We use RoslynCodeDomProvider to compile code that is stored in our server-side database.
We only reference one version of System.Runtime (Version=4.0.0.0, PublicKeyToken=b03f5f7f11d50a3a) in our server-side application but what we reference is not necessarily what gets outputted at compile time due to binding redirects generated by AutoGenerateBindingRedirects.

**Comment by MelbourneDeveloper** [comment]
@Jinhuafei 

-  Is it asp.net core app or the full framework asp.net app?

Full framework.

- Are you using Codedom provider package to generate/compile your own code?

We are using Microsoft.CodeDom.Providers.DotNetCompilerPlatform NuGet package

-  Do you reference another version of system.runtime in your project?

We've tried both not referencing the DLL, and referencing it from multiple places.

When is this going to be fixed?

What more information do you need from us to fix this? There is a bug that you've pointed to in the code. Could you please fix it?

**Comment by BrightLight** [comment]
The [same line](https://github.com/aspnet/RoslynCodeDomProvider/blob/12975a5a1616dbefac02ba8113204b59cba7eeec/src/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/CSharpCompiler.cs#L114) is giving me headaches as well. We have a VSIX that adds a custom project type to Visual Studio. It references the RoslynCodeDomProvider package to generate an assembly. The target framework of that assembly is a project setting. We set the locations of all necessary references in `CompilerParameters.ReferencedAssemblies` (using the "Referenced Assemblies" folder for the desired target framework).
Because of this line (which was added for Bug 913691?) the wrong version of "System.Runtime" is added as a reference and the generated assembly does not target the correct framework version (unless - by chance - you target the latest framework version installed on that machine).
It's also impossible at the moment to generate an assembly that targets .Net Standard.
What exactly is bug 913691? Is `System.Runtime` even necessary if you provide a `CompilerParameters.CoreAssemblyFileName`?

**Comment by StephenMolloy** [comment]
Apologies for the long delay on this one. Matt has moved on to a different team and I am just not getting to look at this repo.

Is this still an issue for folks using version 2 of this package? Has anybody been able to capture a fusion log when the attempted load of System.Runtime.dll happens?

**Comment by StephenMolloy** [comment]
We just published an updated 3.5-preview1 build. Does this issue still affect anybody?

**Comment by StephenMolloy** [comment]
Closing old, inactive bugs.
